java/CMakeLists.txt

```diff
@@ -481,16 +486,23 @@ if(NOT MSVC)
   set_property(TARGET ${ROCKSDB_STATIC_LIB} PROPERTY POSITION_INDEPENDENT_CODE ON)
 endif()
 
-set(ROCKSDBJNI_STATIC_LIB rocksdbjni${ARTIFACT_SUFFIX})
-add_library(${ROCKSDBJNI_STATIC_LIB} ${JNI_NATIVE_SOURCES})
-add_dependencies(${ROCKSDBJNI_STATIC_LIB} rocksdbjni_headers)
-target_link_libraries(${ROCKSDBJNI_STATIC_LIB} ${ROCKSDB_STATIC_LIB} ${LIBS})
+if(NOT CMAKE_SYSTEM_NAME MATCHES "Android")
+  set(ROCKSDBJNI_STATIC_LIB "rocksdbjni${ARTIFACT_SUFFIX}")
+  add_library(${ROCKSDBJNI_STATIC_LIB} ${JNI_NATIVE_SOURCES})
+  add_dependencies(${ROCKSDBJNI_STATIC_LIB} rocksdbjni_headers)
+  target_link_libraries(${ROCKSDBJNI_STATIC_LIB} ${ROCKSDB_STATIC_LIB} ${LIBS})
+endif()
 
 if(NOT MINGW)
-  set(ROCKSDBJNI_SHARED_LIB rocksdbjni-shared${ARTIFACT_SUFFIX})
+  if(CMAKE_SYSTEM_NAME MATCHES "Android")
+    # On Android the shared lib is used in generating the JNI
+    set(ROCKSDBJNI_SHARED_LIB "rocksdbjni${ARTIFACT_SUFFIX}")
```

Comment:
What is the `-shared` omitted for Android when it is still a shared library? If this is a mismatch between this and Environment#getJniLibraryName(String)` or `Environment#getSharedLibraryName(String)`, then I think it would be better to add Android to `Environment.java`, so that the naming becomes more consistent across all platforms.